### PR TITLE
[MRG+1] add repr to report

### PR DESCRIPTION
**Pull request by jasmainak**
small addition. reviews welcome

### CODE REVIEW

File: mne/report.py [hunk 1]
```diff
@@ -174,6 +174,16 @@ def _is_bad_fname(fname):
         return ''
 
 
+def _get_fname(fname):
+    """Get fname without -#-"""
+    if '-#-' in fname:
+        fname = fname.split('-#-')[0]
+    else:
+        fname = op.split(fname)[1]
```
Comment (Eric89GXL) [inline mne/report.py]: Do you mean basename?

Comment (Eric89GXL) [inline mne/report.py]: ...because this will probably be weird for filenames with spaces

File: mne/report.py [hunk 2]
```diff
@@ -814,6 +824,25 @@ def __init__(self, info_fname=None, subjects_dir=None,
...
+            s += ' | %s' % self.title
+        fnames = [_get_fname(f) for f in self.fnames]
+        if len(self.fnames) > 4:
+            s += '\n%s' % '\n'.join(fnames[:2])
+            s += '\n ...\n'
+            s += '\n'.join(fnames[-2:])
+        elif len(self.fnames) > 0:
+            s += '\n%s' % '\n'.join(fnames)
+        s += '\n>'
+        return s
```
Comment (Eric89GXL) [inline mne/report.py]: Can you add a simple test like print(report), ideally hitting each branch here?

Comment (agramfort) [inline mne/report.py]: can you paste to show how this renders?

Comment (jasmainak) [inline mne/report.py]: If you run [this gist](https://gist.github.com/jasmainak/8b2617c7fb4c37f85a363fa9784621b2) and then type in `report` in `ipython`, you will see something like this:

```sh
In [6]: report
Out[6]: 
<Report | 7 items | MNE Report for ...ples/MNE-sample-data
 ... sample_audvis_filt-0-40_raw-eve.fif
 ... sample_audvis_eog-eve.fif
 ...
 ... sample_audvis_raw.fif
 ... Left Auditory
>
```

**Comment by agramfort** [comment]
LGTM

@Eric89GXL merge if you're happy too

**Comment by Eric89GXL** [comment]
Comments not addressed yet

File: mne/report.py (hunk 1 above)
Comment (jasmainak) [inline mne/report.py]: It's the same according to the documentation here: https://docs.python.org/2/library/os.path.html. I tried them for some filenames with spaces and it seemed to give the same result. I changed it anyway :)

File: mne/report.py (hunk 2 above)
Comment (jasmainak) [inline mne/report.py]: I added a couple more lines in the tests and verified that I was hitting all the branches

File: mne/report.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/report.py]: Ahh you're right, I was thinking of `str.split`

**Comment by Eric89GXL** [comment]
Codecov thinks there are still some branches missed, are you sure you got them all?

**Comment by Eric89GXL** [comment]
It's not a big deal if you don't get all of them, but you said you got them all so I'm not sure what's going on...

**Comment by jasmainak** [comment]
@Eric89GXL : this report is not the latest one: https://codecov.io/gh/mne-tools/mne-python/pull/3492/compare ? I'm not sure how exactly to interpret the codecov report. I see all greens and one yellow on the left ... so I was assuming everything is okay?

**Comment by Eric89GXL** [comment]
Weird, codecov is supposed to update its posts. Sorry for the noise

**Pull request merged**